nemo/src/io/parser.rs

```diff
@@ -35,6 +35,18 @@ pub fn parse_program(input: impl AsRef<str>) -> Result<Program, Error> {
     Ok(program)
 }
 
+/// Parse a single fact in the given `input`-String and return a [`Program`].
+///
+/// The program will be parsed and checked for unsupported features.
+///
+/// # Error
+/// Returns an appropriate [`Error`] variant on parsing and feature check issues.
+pub fn parse_fact(mut input: String) -> Result<Fact, Error> {
+    input += ".";
+    let fact = all_input_consumed(RuleParser::new().parse_fact())(input.as_str())?;
+    Ok(fact)
+}
```

Comment:
This is fine in principle, but using a fresh parser has the drawback that the fact cannot use, e.g., prefixed names that would need resolving.

Reply:
Yeah its a pretty lazy solution.

Maybe `parse_program` should return the `RuleParser` as well? Then I could use it to parse the additional fact